src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+elseif(CMAKE_THREAD_LIBS_INIT)
+  list(APPEND LIBS ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
```

Comment:
Hmm, okay, I guess I should extend the test around the -pthread addition to c(xx)flags rather than use this module (also, I did not see that there was an internal module to detect it).

Thanks, I'll retract this PR.